src/boot/TranslationAPIProvider.js

```diff
@@ -0,0 +1,28 @@
+/* @flow */
+import { PureComponent } from 'react';
+import { injectIntl, intlShape } from 'react-intl';
+
+import type { ChildrenArray } from '../types';
+
+type Props = {
+  children: ChildrenArray<*>,
+  intl: intlShape.isRequired,
+};
+
+class TranslationAPIProvider extends PureComponent<Props> {
+  static childContextTypes = {
+    _: () => '',
```

Comment:
I've never really quite understood what `contextTypes` is supposed to do. @borisyankov , can you explain why it's there and why the types all seem to be `() => {}`?

I did look at https://reactjs.org/docs/legacy-context.html which seems to be the relevant documentation, but the types there are all from `PropTypes`.